src/klement.jl

```diff
@@ -0,0 +1,62 @@
+"""
+```julia
+Klement()
+```
+
+A low-overhead implementation of [Klement](https://jatm.com.br/jatm/article/view/373).
+This method is non-allocating on scalar and static array problems.
+"""
+struct Klement <: AbstractSimpleNonlinearSolveAlgorithm end
+
+function SciMLBase.solve(prob::NonlinearProblem,
+                         alg::Klement, args...; abstol = nothing,
+                         reltol = nothing,
+                         maxiters = 1000, kwargs...)
+    f = Base.Fix2(prob.f, prob.p)
+    x = float(prob.u0)
+    fₙ = f(x)
+    T = eltype(x)
+    J = ArrayInterfaceCore.zeromatrix(x) + I
+
+    if SciMLBase.isinplace(prob)
+        error("Klement currently only supports out-of-place nonlinear problems")
+    end
+
+    atol = abstol !== nothing ? abstol :
+           real(oneunit(eltype(T))) * (eps(real(one(eltype(T)))))^(4 // 5)
+    rtol = reltol !== nothing ? reltol : eps(real(one(eltype(T))))^(4 // 5)
+
+    xₙ = x
+    xₙ₋₁ = x
+    fₙ₋₁ = fₙ
+    for _ in 1:maxiters
+        xₙ = xₙ₋₁ - inv(J) * fₙ₋₁
```

Comment:
Always use the matrix factorization object to solve.

Reply:
I missed that. Instead, just `\` here.

Reply:
Or `F = lu(J)`, and the singularity check is then `F.factors[end, end] < singular_tol`.